docs/DeveloperGuide.md

```diff
@@ -1,30 +1,126 @@
 # Developer Guide
 
+---
+Welcome to the PlanPal Developer Guide! Thank you for taking an interest in the behind-the-scenes work of our product. We hope this document proves informative and useful for your work!
+
+## Table of Contents
+{to be inserted at a later date}
+
+---
+
 ## Acknowledgements
 
+
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Setting up, getting started
+1. Java 17 is required for PlanPal to function properly, please make sure your device has Java 17 installed. If not, you may download it [here](). If you need further assistance installing Java 17, you may refer to the [Installation Guide.]()
+2. Once you have ascertained that your device has Java 17 installed, you may download the `.jar file` [here]().
+insert SS of release page
+
+Click on `PlanPal.jar` to automatically download the file. Place the file in a folder your choice.
+
+
+
+## Design
+
+---
+### Architecture
+![Architecture.drawio.png](Images%2FArchitecture.drawio.png)
+
+The **Architecture Diagram** given above explains the high-level design of the program. Given below is a quick overview of the main components.
+
+### Main Components
+- `PlanPal`: Main entry of the program, initialises and connects the components
+- `UI`: In charge of printing messages
+- `Logic`: Determines the command to execute
+- `Storage`: Read and write data from hard disk
+- `Command`: Specific commands for execution
+
+### Program Flow
+- The `PlanPal` controls access to the `UI`, `Parser`, and `FileManager` (Storage) components of the app.
+- The user will send input to `PlanPal`.
+- `PlanPal` will parse the input to produce a `Command`, which will hook back into `PlanPal` to gain access to the other components when running
+
+---
+## Implementation
+---
+
+The class diagram below represents the contact book system
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+![ContactClassDiagram.jpg](Images/ContactClassDiagram.jpg)
 
+Currently, only the contact features have been illustrated.
+{to add other implementations for expense and activity managers}
 
+## Contact Manager Features
+
+### Add Command
+The sequence diagram below illustrates the process for resolving the "add" command.
+![AddContact.drawio.png](Images%2FAddContact.drawio.png)
```

Comment:
Should the font size of the words in the sequence diagram be consistent with the DG as mentioned in the lecture notes?
![image](https://github.com/user-attachments/assets/d24f4ba5-5fe3-486a-921d-a492a9a0809d)